app/core/tests/test_api.py

```diff
@@ -408,6 +409,39 @@ def test_create_soc_major(auth_client):
     assert res.data["title"] == payload["title"]
 
 
+def test_create_soc_minor(auth_client):
+    """Test that we can create a soc minor"""
+
+    payload = {
+        "occ_code": "33-3333",
+        "title": "Test soc minor",
+    }
+    res = auth_client.post(SOC_MINORS_URL, payload)
+    assert res.status_code == status.HTTP_201_CREATED
+    assert res.data["title"] == payload["title"]
+    assert res.data["occ_code"] == payload["occ_code"]
+
+
+def test_soc_minor_soc_major_relationship(auth_client, soc_minor, soc_major):
+    res = auth_client.patch(
+        SOC_MINORS_URL + str(soc_minor.pk) + "/set_soc_major/",
+        {"soc_major": soc_major.pk},
+    )
+    assert res.status_code == status.HTTP_202_ACCEPTED
```

Comment:
The custom viewset action is not really needed here and adds complexity to the API.